lib/components/notifications-component.js

```diff
@@ -15,9 +16,23 @@ export default class NotificationsComponent extends BaseContainer {
 		let text = this.driver.findElement( by.css( '.wpnc__notes' ) ).getText();
 		return text;
 	}
+
 	selectCommentByText( commentText ) {
-		const commentSelector = by.xpath( `//div[normalize-space(text())='${commentText}']` );
-		return driverHelper.clickWhenClickable( this.driver, commentSelector );
+		const self = this;
+		const commentSelector = by.css( 'div > div.wpnc__text-summary' );
+		let comments = this.driver.findElements( by.css( 'div.wpnc__excerpt' ) );
+		this.driver.wait( until.elementLocated( commentSelector ), self.explicitWaitMS, 'Could not locate comment exerpt.' );
+		webdriver.promise.filter( comments, function( comment ) {
+			return comment.getText().then( function( text ) {
+				return text === commentText;
+			} );
+		} ).then( function( filteredComments ) {
+			if ( filteredComments.length === 1 ) {
+				return driverHelper.clickWhenClickable( self.driver, commentSelector );
```

Comment:
This filter just tells me if the comment I'm looking for exists _somewhere_ in the list, and then it clicks on the first one regardless of whether that was the match or not.  Instead of returning `text === commentText` can we return the `comment` WebElement itself, and then use that in the `clickWhenClickable` action?  I haven't tested that to see if it works or not.

BTW, the method I'm using to force this scenario is just modifying the wp-notifications-spec.js file to write the comment with the randomly-generated phrase, but then in the second half of the spec where the other user logs in I override `this.comment` with a value I know exists elsewhere in the list.